File: ui/src/bridges/startBridge.ts
import type { QtSignal } from "@/types/channel";

export interface StartBridge {
  // signals
  onLoginSucceeded: QtSignal<(nickname: string) => void>;
  onLoginFailed: QtSignal<(error: string) => void>;

  // slots
  callVerifyNickname: (nickname: string) => void;
}

File: src/client/Bridges/StartBridge.cpp
#include "StartBridge.hpp"

void StartBridge::hasLoginSucceeded(const QString& nickname) {
  qDebug() << "StartBridge: hasLoginSucceeded called with nickname:" << nickname;
  emit onLoginSucceeded(nickname);
}

void StartBridge::hasLoginFailed(const QString& error) {
  qDebug() << "StartBridge: hasLoginFailed called.";
  emit onLoginFailed(error);
}

void StartBridge::callVerifyNickname(const QString& nickname) {
  qDebug() << "StartBridge: callVerifyNickname called with nickname:" << nickname;
  emit requestVerifyNickname(nickname);
}
